dot/rpc/subscription/websocket.go

```diff
@@ -159,25 +162,24 @@ func (c *WSConn) initStorageChangeListener(reqID float64, params interface{}) (L
 		wsconn: c,
 	}
 
-	pA, ok := params.([]interface{})
-	if !ok {
-		return nil, fmt.Errorf("unknown parameter type")
-	}
-	for _, param := range pA {
-		switch p := param.(type) {
-		case []interface{}:
-			for _, pp := range param.([]interface{}) {
-				data, ok := pp.(string)
-				if !ok {
-					return nil, fmt.Errorf("unknown parameter type")
-				}
-				stgobs.filter[data] = []byte{}
+	switch filter := params.(type) {
+	case []string:
+		for _, key := range filter {
+			stgobs.filter[key] = []byte{}
+		}
+	case []interface{}:
+		for _, interfaceKey := range params.([]interface{}) {
```

Comment:
I think that would work right?

```suggestion
		for _, interfaceKey := range filter {
```